### PR TITLE
interceptor improvements, drop php 7.3

### PR DESCRIPTION
**Pull request by splittingred**
## What?

* Add `getFullyQualifiedMethodName` and `getExpectedResponseMessageClass` to base Client Interceptor class
* Drop php 7.3 support (EOL)

### CODE REVIEW

**Review by lord2800 — APPROVED** [review]

**Review by lord2800 — APPROVED** [review]

File: src/Grphp/Client/Interceptors/Base.php [hunk 1]
```diff
@@ -72,6 +72,46 @@ public function setRequest(&$request)
         $this->request = $request;
     }
 
+    /**
+     * Gets the fully qualified method name, e.g. grphp.catalog.products/GetProduct
+     *
+     * @return string
+     * @throws StubNotFoundException
+     */
+    public function getFullyQualifiedMethodName(): string
```
Comment (aleachjr) [inline src/Grphp/Client/Interceptors/Base.php:81]: do we want this to be the fully qualified name, or the fully qualified name that matches expected responses? 

`grphp.catalog.products/GetProduct` vs `grphp.catalog.products/getProduct`

In my local version had to lcfirst the method to match

File: src/Grphp/Client/Interceptors/Base.php [hunk 2]
```diff
@@ -72,6 +72,46 @@ public function setRequest(&$request)
...
+    {
+        $methodName = $this->getMethod();
+        $stub = $this->getStub();
+        if (empty($stub)) {
+            throw new StubNotFoundException("Stub not found for $methodName");
+        }
+
+        $responseMessages = $stub->getExpectedResponseMessages();
+
+        if (!array_key_exists($methodName, $responseMessages)) {
```
Comment (aleachjr) [inline src/Grphp/Client/Interceptors/Base.php:109]: this is where i needed the lcfirst to match

File: src/Grphp/Client/Interceptors/Base.php (hunk 1 above)
Comment (splittingred) [reply src/Grphp/Client/Interceptors/Base.php:81]: Yep, you're right, TAL

Comment (splittingred) [reply src/Grphp/Client/Interceptors/Base.php:81]: So it looks like our generator actually generators them with the lowercase, e.g:

```
        'addAppCategory' => '\Bigcommerce\Rpc\Apps\AppCategoryResp',
        'removeAppCategory' => '\Bigcommerce\Rpc\Apps\AppCategoryResp',
```

**Review by aleachjr — APPROVED** [review]

**Comment by aleachjr** [comment]
Thanks for adding this in!

**Pull request merged**